### PR TITLE
SLING-7988 - Allow Analyzer include/exclude tasks by id, in order to enable specific Tasks in the Maven plugin

### PR DESCRIPTION
**Pull request by simonetripodi**
Hi there @bosschaert,
please have a look at the initial implementation of my proposal.

I moved few utility static methods from Analyzer class to a proper dedicated Task provider implementation, in order to have separation of concerns and better unit testing.

I also implemented a new unit test, all old tests still pass.

It also includes a little logging improvement in order to make the developers/users better understand what's going under the hood

### CODE REVIEW

File: src/main/java/org/apache/sling/feature/analyser/task/AnalyzerTaskProvider.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.ServiceLoader;
+import java.util.Set;
+
+import org.apache.sling.feature.analyser.Analyser;
+
+public final class AnalyzerTaskProvider {
```
Comment (bosschaert) [inline src/main/java/org/apache/sling/feature/analyser/task/AnalyzerTaskProvider.java]: Does this new class need to be public? If it's only used from within the same package then 'package' protected should also be possible, right?

File: src/main/java/org/apache/sling/feature/analyser/task/AnalyzerTaskProvider.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+    private AnalyzerTaskProvider() {
+        // this class must not be instantiated directly
+    }
+
+    public static AnalyserTask[] getTasks() {
```
Comment (bosschaert) [inline src/main/java/org/apache/sling/feature/analyser/task/AnalyzerTaskProvider.java]: The methods don't need to be public either, I guess?

Comment (simonetripodi) [reply src/main/java/org/apache/sling/feature/analyser/task/AnalyzerTaskProvider.java]: Uhm I implemented the provider under the `.task` package (where the `AnalyserTask` is defined), that means methods would not be visible to the `Analyzer` class...

Comment (bosschaert) [reply src/main/java/org/apache/sling/feature/analyser/task/AnalyzerTaskProvider.java]: Ah, sorry, missed that. 
Thanks for clarifying, @simonetripodi !

Comment (simonetripodi) [reply src/main/java/org/apache/sling/feature/analyser/task/AnalyzerTaskProvider.java]: My pleasure, always glad to cooperate! :)

**Comment by bosschaert** [comment]
Thanks @simonetripodi your pull request is merged!

**Comment by simonetripodi** [comment]
Priceless @bosschaert , thanks for taking care of it! :)

**Pull request merged**